src/dependency-manager/src/manager.ts

```diff
@@ -30,9 +32,10 @@ export default abstract class DependencyManager {
     this.__graph_cache = {};
   }
 
-  async init(environment_config?: EnvironmentConfig): Promise<void> {
+  async init(environment_config?: EnvironmentConfig, values_dictionary: Dictionary<Dictionary<string>> = {}): Promise<void> {
```

Comment:
I've been wondering is it safe to assign a mutable object in the constructor? Not that we modify values_dictionary in the class object, but I'm worried this is an anti-pattern if javascript behaves the same way as python.

Reply:
Did you have any alternatives in mind? We've made it like a year without messing up the environment config that's stored in the same way, so I wasn't too worried about it

Reply:
I meant the constructor arg. I don't think you're supposed to set mutable defaults. Def an anti pattern in python.

Arg = values_dictionary?
this.values_dictionary = values_dictionary || {}

Reply:
Ignore me. Different than python.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#Evaluated_at_call_time